src/api/departmentsApi.js

```diff
@@ -0,0 +1,22 @@
+import { ApiRequest } from './apiService';
+import { DEPARTMENTS_URL } from '../constants/url';
+
+class DepartmentApi {
+  get(params) {
+    return ApiRequest.get(DEPARTMENTS_URL, params);
+  }
+  getOne(id) {
+    return ApiRequest.get(`${DEPARTMENTS_URL}/${id}`);
+  }
+  add(data) {
+    return ApiRequest.post(DEPARTMENTS_URL, data);
+  }
+  delete(id) {
+    return ApiRequest.delete(`${DEPARTMENTS_URL}/${id}`);
+  }
+  update(id, data) {
+    return ApiRequest.patch(`${DEPARTMENTS_URL}/${id}`, data);
+  }
```

Comment:
you can utilize the class extensions
e.g.you can pass baseUrl as constructor parameter in BaseAPI